### PR TITLE
Add --skip parameter to skip outputs during postprocessing

### LINKED ISSUE #1058: How about optionally disable ALFF and ReHo?
## Summary

When we only want to calculate functional connectivity matrices, ALFF and ReHo calculations are not used and take extra time to execute. So, I am wondering if it is possible to optionally disable these calculations. Or, more flexibly, let users select outputs.

## Additional details

The pipeline will be much more flexible then.

## Next steps

Maybe add options `--ouputs`, or parameters `--disable-alff`/`--disable-reho`?

### PR DESCRIPTION
**Pull request by psychelzh**
## Changes proposed in this pull request

Closes #1058.

 This PR adds a `--skip` parameter to XCP-D, allowing users to selectively skip specific outputs during postprocessing to save computation time.

### Features

**New `--skip` parameter with 4 options:**
- `alff` - Skip ALFF calculation (only when bandpass filtering is enabled)
- `reho` - Skip ReHo calculation
- `parcellation` - Skip parcellation and time series extraction (also skips connectivity)
- `connectivity` - Skip connectivity calculations only (preserves parcellation for time series)

**Key behavior:**
- Skipping `parcellation` automatically skips `connectivity` (dependency enforcement)
- Skipping `connectivity` alone sets `correlation_lengths = []` to disable correlation matrices while preserving time series extraction

**Bug fix:**
- Fixed `correlation_lengths` handling to work with all values (numeric, `'all'`, or mixed), not just `'all'`

### Files Changed

- **Core:** `parser.py`, `config.py`, `workflows/bold/nifti.py`, `workflows/bold/cifti.py`
- **Tests:** `test_skip_parameter.py` (337 lines, 17 tests), `test_cli_run.py`
- **Docs:** `usage.rst`, `workflows.rst`, `outputs.rst`

### Testing

- ✅ All skip options tested individually and in combinations
- ✅ Edge cases: empty atlases, various `correlation_lengths` values
- ✅ Dependency enforcement validated
- ✅ Backward compatibility maintained

---

### Updated Files

**`docs/usage.rst`**
- Added `--skip` parameter description with usage examples
- Documented dependency relationships and warnings

**`docs/workflows.rst`**
- Explained how skip options affect workflow execution
- Documented conditional node behavior

**`docs/outputs.rst`**
- Clarified which outputs are conditional based on skip settings

### Review Focus

- Parameter help text clarity (`parser.py` lines 329-346)
- Validation logic correctness (`parser.py` lines 1350-1385)
- Workflow integration in NIfTI/CIFTI workflows

---

<sub>*This PR was developed with AI assistance (GitHub Copilot).*</sub>

---

> [!NOTE]
> Adds a unified --skip flag (alff|reho|parcellation|connectivity) with workflow/output gating, deprecates --skip-parcellation, and updates validation, docs, and tests.
> 
> - **CLI/Validation**:
>   - Add `--skip` (`alff`, `reho`, `parcellation`, `connectivity`) with help text and parsing in `xcp_d/cli/parser.py`.
>   - Map deprecated `--skip-parcellation` to `--skip parcellation`; warn and keep backward compatibility.
>   - Enforce dependencies: `parcellation` ⇒ auto-add `connectivity`, clear `correlation_lengths`; allow `--create-matrices` only when connectivity is not run in `linc`.
>   - Normalize mode defaults (e.g., `nichart` `correlation_lengths` now list).
> - **Workflows/Outputs**:
>   - Gate ALFF and ReHo: compute/connect only when not skipped; ALFF auto-skipped if bandpass disabled (`bold/nifti.py`, `bold/cifti.py`).
>   - Propagate `skip_reho`/`skip_alff` into connectivity workflows; conditionally parcellate metrics (`bold/connectivity.py`).
>   - Suppress sinks for skipped metrics (e.g., no ReHo/ALFF outputs) in `bold/outputs.py`.
> - **Config**:
>   - Add `workflow.skip_outputs`; widen `set_start_method` exception handling in `config.py`.
> - **Docs**:
>   - Document `--skip` options and deprecate `--skip-parcellation` in `docs/usage.rst`, `docs/workflows.rst`, `docs/outputs.rst`.
> - **Tests/Tooling**:
>   - Add/adjust unit and integration tests for `--skip` behavior; include new expected output listing; update existing tests to use `--skip`.
>   - Switch pytest env var to `[tool.pytest_env]` in `pyproject.toml`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 986dff939886e52a8b090a687ebb961e3be42aec. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by tsalo** [comment]
@psychelzh I've only briefly glanced at it, but this is looking really good. I will try to review it in depth soon. In the meantime, can you (1) modify one of the integration tests in `test_cli.py` to skip ReHo creation (you'll need to update the test's expected outputs as well) and (2) deprecate `--skip-parcellation` in favor of `--skip parcellation`? The tests in `test_skip_integration.py` don't seem to be triggered, and we try to cover all of the parameters in the integration tests within `test_cli.py` rather than add new ones, since they take so long.

File: xcp_d/cli/parser.py [hunk 1]
```diff
@@ -1327,6 +1367,51 @@ def _validate_parameters(opts, build_log, parser):
...
+                'Parcellation will still be performed to extract time series, '
+                'but no correlation matrices will be computed.'
+            )
+            # Set correlation_lengths to empty to skip connectivity calculations
+            # while still performing parcellation
+            opts.correlation_lengths = []
+
+        # Warn if ALFF is skipped but bandpass filter is disabled
+        if 'alff' in opts.skip_outputs and not opts.bandpass_filter:
+            build_log.warning('Skipping ALFF has no effect when bandpass filtering is disabled.')
```
Comment (tsalo) [inline xcp_d/cli/parser.py]: I don't think there's any need for this warning.

File: xcp_d/workflows/bold/connectivity.py [hunk 2]
```diff
@@ -567,7 +580,7 @@ def init_functional_connectivity_cifti_wf(mem_gb, exact_scans, name='connectivit
             ]),
         ])  # fmt:skip
 
-    if bandpass_filter:
+    if bandpass_filter and not skip_alff:
```
[the remark this answers is not in the document]
Comment (psychelzh) [reply xcp_d/workflows/bold/connectivity.py]: Done.

File: xcp_d/workflows/bold/cifti.py [hunk 3]
```diff
@@ -270,33 +270,41 @@ def init_postprocess_cifti_wf(
         ])  # fmt:skip
 
     if bandpass_filter:
-        alff_wf = init_alff_wf(name_source=bold_file, TR=TR, mem_gb=mem_gbx)
+        # Skip ALFF calculation if requested
+        skip_alff = 'alff' in config.workflow.skip_outputs
+        if not skip_alff:
```
[the remark this answers is not in the document]
Comment (psychelzh) [reply xcp_d/workflows/bold/cifti.py]: Now I have defined `skip_alff` before the condition, which is more aligned with `skip_reho`.
